docs/proposed/recovery.rst

```diff
@@ -0,0 +1,97 @@
+.. -*- coding: utf-8 -*-
+
+.. _snapshots:
+
+Audience
+========
+
+This document is aimed at programmers working on integrating with magic-folder.
+
+
+Motivation
+==========
+
+It is useful to be able to back up enough information in order to recover a magic-folder on a new device.
+This document describes both the process of gathering the informatino to back up and using it to re-create a similar magic-folder.
+
+
+Overview
+========
+
+A magic-folder consists of a Collective DMD containing sub-directories pointing at the read-capabilities of one or more participant's Personal DMDs.
+If this magic-folder instance is the "admin" instance, it will contain a write-capability for the Collective.
+It will also contain a write-capablity corresponding to the Personal DMD.
+
+The rough process of "recovery" is:
+
+- create a new, fresh Magic Folder on the new instance
+- add the "recovery" instance as a participant in this new Magic Folder
+
+The new instance's local magic-path will fill up with all the files in the old instance as the downloader gets to them.
+In a usual case of "recovery" from e.g. a lost or broken device, this participant would not produce new Snapshots.
+However, if it does start producing new Snapshots the Magic Folder will continue to synchronize and detect conflicts as normal.
+
+**Note:** this document is only concerned with single-participant Magic Folders for now and does not cover recovery of other participants.
+
+
+Gathering the Information
+-------------------------
+
+In order to recover a Magic Folder on a new device you will need some information about it:
+
+- the Collective DMD (either write-capability or read-capability)
+- the Personal DMD (usually a write-capability but you could choose to only save the read-capability corresponding to it)
```

Comment:
The point of being ambiguous here is that it's up to the integrator to decide that.

I think GridSync probably _should_ save it, since it's advertising it as "a backup" even though it only wants to use the read-capability with magic-folder. No user-data will be lost either way though.

Reply:
I am inclined to say that we should be opinionated on how people should integrate with us, as much as possible. Particularly around security sensitive things.